core/model-api/src/main/java/org/eclipse/rdf4j/model/base/AbstractLiteral.java

```diff
@@ -175,6 +333,25 @@ public BigDecimal decimalValue() {
 		return value(BigDecimal::new);
 	}
 
+	@Override
+	public TemporalAccessor temporalValue() throws DateTimeException {
+		return value(label -> {
+
+			final TemporalAccessor value = label.startsWith("--") ? DASH_FORMATTER.parse(label)
+					: label.length() >= 8 && label.charAt(2) == ':' ? OFFSET_TIME_FORMATTER.parse(label)
+							: DATETIME_FORMATTER.parse(label);
```

Comment:
Can we not use nested/chained ternaries? YMMV, but I find them hard to read. Might be improved with linebreaks in different places, but I think an if-else is easier on the eye (if a little more verbose):

```java
TemporalAccessor value;
if (label.startsWith("--")) {
    value = DASH_FORMATTER.parse(label);
} else if (label.length() >= 8 && label.charAt(2) == ':' ) {
   value = OFFSET_TIME_FORMATTER.parse(label);
} else {
  value = DATETIME_FORMATTER.parse(label);
}
```

or perhaps more clean:

```java
final TemporalAccessor value = getFormatter(label).parse();

... 

private static DateFormatter getFormatter(String label) {
    if(label.startsWith("--")) {
       return DASH_FORMATTER;
   } 
   if (label.length() >= 8 && label.charAt(2) == ":") {
       return OFFSET_TIME_FORMATTER;     
   }
   return DATETIME_FORMATTER;
}
```

Reply:
Done.

Reply:
Great, thanks!